experimental/instrumentation-fetch/src/instrumentation.ts

```diff
@@ -0,0 +1,162 @@
+import { BaseInstrumentation, faro, globalObject } from '@grafana/faro-core';
+
+import {
+  eventDomain,
+  fetchGlobalObjectKey,
+  originalFetch,
+  originalFetchGlobalObjectKey,
+  parseHeaders,
+  rejectedFetchEventName,
+  resolvedFetchEventName,
+  responseProperties,
+  VERSION,
+} from './constants';
+import type { FetchInstrumentationOptions } from './types';
+
+export class FetchInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-fetch';
+  readonly version = VERSION;
+  private ignoredUrls: FetchInstrumentationOptions['ignoredUrls'];
+
+  constructor(private options?: FetchInstrumentationOptions) {
+    super();
+  }
+
+  /**
+   * Initialize fetch instrumentation - globalObject.fetch becomes instrumented fetch, assign original fetch to globalObject.originalFetch
+   */
+  initialize(): void {
+    this.internalLogger.info('Initializing fetch instrumentation');
+    this.ignoredUrls = this.options?.ignoredUrls ?? this.getTransportIgnoreUrls();
+
+    Object.defineProperty(globalObject, fetchGlobalObjectKey, {
+      configurable: true,
+      enumerable: false,
+      writable: false,
+      value: this.instrumentFetch(),
+    });
+
+    Object.defineProperty(globalObject, originalFetchGlobalObjectKey, {
+      configurable: true,
+      enumerable: false,
+      writable: false,
+      value: originalFetch,
+    });
+  }
+
+  /**
+   * Generate a unique request id object for each fetch request
+   */
+  private requestId(idOnly?: boolean): Record<string, string> | string {
+    const requestId = (faro.config.session?.id ?? faro.config.user?.id) + Date.now().toString();
+    return idOnly ? requestId : { request_id: requestId };
+  }
+
+  /**
+   * Get the list of ignored urls from all transports
+   */
+  private getTransportIgnoreUrls(): Array<string | RegExp> {
+    return faro.transports.transports?.flatMap((transport) => transport.getIgnoreUrls());
+  }
+
+  /**
+   * Get the list of ignored urls from all transports
+   */
+  getIgnoredUrls(): Array<string | RegExp> {
+    return this.ignoredUrls ?? [];
+  }
+
+  /**
+   * Instrument fetch with Faro
+   */
+  private instrumentFetch(): typeof fetch {
+    const instrumentation = this;
+    return function fetch(this: typeof globalObject, input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
```

Comment:
Suggestion: 
Instead of passing `this` as a parameter you can either use `bind` or an arrow function when you create the respective fetch object  (`value:...`).

for example somethig like:
```ts
{
  value: () => this.instrumentFetch(),
}
```

or 

```ts
{
  value: () => this.instrumentFetch(),
}
```